src/Haskell/Ide/Engine/Transport/JsonStdio.hs

```diff
@@ -76,9 +76,9 @@ run dispatcherProc cin = flip E.catches handlers $ do
           , docVersionTVar = versionTVar
           }
 
-    _opid <- forkIO $ outWriter rout
-    _rpid <- forkIO $ reactor rout
     dispatcherProc dispatcherEnv
+    race_ (outWriter rout) (reactor rout)
+
```

Comment:
Since both are infinitely looping, I don't think it matters much in practice. But with `race_` , if one thread exits normally, the other is killed, which is behaviour we probably want.

I could make the type signatures `forall void. IO void` to make this explicit though. Then we could get rid of the unreachable `return 0`.